custom-recipes/azure-cognitive-services-nlp-key-phrases-extraction/recipe.py

```diff
@@ -1,62 +1,98 @@
-import logging
-import time
-import json
+# -*- coding: utf-8 -*-
+from typing import List, Dict, AnyStr
+
+from retry import retry
+from ratelimit import limits, RateLimitException
+
 import dataiku
-from dataiku.customrecipe import *
-from dku_azure_cs import *
-from common import *
+from dataiku.customrecipe import get_recipe_config, get_input_names_for_role, get_output_names_for_role
 
-#==============================================================================
-# SETUP
-#==============================================================================
+from plugin_io_utils import ErrorHandlingEnum, validate_column_input, set_column_description
+from azure_nlp_api_client import API_EXCEPTIONS, batch_api_response_parser, get_client
+from api_parallelizer import api_parallelizer
+from azure_nlp_api_formatting import KeyPhraseExtractionAPIFormatter
 
-logging.basicConfig(level=logging.INFO, format='[azure-cognitive-services plugin] %(levelname)s - %(message)s')
 
-connection_info = get_recipe_config().get('connection_info')
-text_column = get_recipe_config().get('text_column')
-language = get_recipe_config().get('language')
-should_output_raw_results = get_recipe_config().get('should_output_raw_results')
+# ==============================================================================
+# SETUP
+# ==============================================================================
 
-input_dataset_name = get_input_names_for_role('input_dataset')[0]
+api_configuration_preset = get_recipe_config().get("api_configuration_preset")
+api_quota_rate_limit = api_configuration_preset.get("api_quota_rate_limit")
+api_quota_period = api_configuration_preset.get("api_quota_period")
+parallel_workers = api_configuration_preset.get("parallel_workers")
+batch_size = api_configuration_preset.get("batch_size")
+text_column = get_recipe_config().get("text_column")
+text_language = get_recipe_config().get("language")
+language_column = get_recipe_config().get("language_column")
+num_key_phrases = int(get_recipe_config().get("num_key_phrases"))
+error_handling = ErrorHandlingEnum[get_recipe_config().get("error_handling")]
+
+input_dataset_name = get_input_names_for_role("input_dataset")[0]
 input_dataset = dataiku.Dataset(input_dataset_name)
 input_schema = input_dataset.read_schema()
-input_columns_names = [col['name'] for col in input_schema]
-keyphrases_column = generate_unique('keyphrases', input_columns_names)
+input_columns_names = [col["name"] for col in input_schema]
 
-output_dataset_name = get_output_names_for_role('output_dataset')[0]
+output_dataset_name = get_output_names_for_role("output_dataset")[0]
 output_dataset = dataiku.Dataset(output_dataset_name)
 
```

Comment:
Maybe you could also validate the language_column
```suggestion
if text_language == "language_column":
    validate_column_input(language_column, input_columns_names)
```